test/test_readonly_task.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
```

Comment:
Sorry.... Just realized that this was a new file.

Test files have a 1:1 relationship with files in lib/. Keep that relationship.

In other words, these tests should be in `test_task_context.rb`. Just put them in a describe block to scope it

```
describe "read_only" do
end
```